File: Telegram/TelegramApp/src/game/gameplay/DamageEvent.ts
// Ported from Assets/PixelPlanets/StellarBreaker/Scripts/Gameplay/DamageEvent.cs
import type { BigNumber } from '../core/BigNumber'

/** Logic payload for a single damage application, used to drive floating damage numbers. */
export interface DamageEvent {
  amount: BigNumber
  isCrit: boolean
}

File: Unity/Assets/PixelPlanets/StellarBreaker/Scripts/Gameplay/DamageEvent.cs
using StellarBreaker.Core;

namespace StellarBreaker.Gameplay
{
    /// <summary>
    /// Logic payload for a single damage application. The View layer subscribes to
    /// TapController.OnDamageDealt and spawns a floating number from this.
    /// (Screen position is a View concern; crit comes online with skills.)
    /// </summary>
    public readonly struct DamageEvent
    {
        public readonly BigNumber Amount;
        public readonly bool      IsCrit;

        public DamageEvent(BigNumber amount, bool isCrit = false)
        {
            Amount = amount;
            IsCrit = isCrit;
        }
    }
}
